pyproject.toml

```diff
@@ -212,6 +212,7 @@ ignore = [
 [tool.pyright]
 # These are tentative
 # include = ["**/*.py"]
-pythonVersion = "3.8" # check no python > 3.8 features are used
-pythonPlatform = "All"
+pythonVersion = "3.10"
 typeCheckingMode = "strict"
+useLibraryCodeForTypes = true
+reportGeneralTypeIssues = true
```

Comment:
What do these do?

Reply:
pyright.useLibraryCodeForTypes [boolean]: This setting is deprecated in favor of python.analysis.useLibraryCodeForTypes. It will be removed at a future time.

python.analysis.useLibraryCodeForTypes [boolean]: Determines whether pyright reads, parses and analyzes library code to extract type information in the absence of type stub files. Type information will typically be incomplete. We recommend using type stubs where possible. The default value for this option is true.

may help with libraries that don't give us types, like pymacaroons


reportGeneralTypeIssues is actuallyu "error" by default, so maybe I don;'t need this setting.


to be honest, I'm just poking a black cat in the black room in the dark when it comes to pyright settings 🐈‍⬛